### DISCUSSION TITLE
Trouble Adding R454B to CoolProp

### DISCUSSION THREAD

**Discussion by KoushaMirjavadi** [CoolProp/CoolProp#2423] [Q&A]
I am having number of problems when I want to add R454B to CoolProp, I would appreciate your help 
First, I created a JSON file for R454B, then I downloaded CoolProp from GitHub version 6.6.0 then I copied and pasted R454B to dev/fluids.
The main problem now is that when I want to rebuild the library and install it on my python, I encounter many errors. I am unable to rebuild the library, 
I have tried CMake and it gets stuck on the first step when I try to rebuild the library, bellow is the response I get on command window : 

"C:\Users\kmirj\Desktop\New Refrigerant\CoolProp\build>cmake --build . --config Release
MSBuild version 17.11.9+a69bbaaf5 for .NET Framework

  1>Checking Build System
  Building Custom Rule C:/Users/kmirj/Desktop/New Refrigerant/CoolProp/CMakeLists.txt" 

It never goes to the second step; I also have used SWIG as another option (ChatGPT suggestion) but same thing happens. 

I also tried to use the way that was mentioned on the website (not exactly) I tried to push my JSON file into the library (below is my code) :
          import CoolProp.CoolProp as CP
          import json
          
          # Path to the custom fluid JSON file
          json_file_path = r"C:\Users\kmirj\Desktop\New Refrigerant\CoolProp\dev\fluids\R454B.json"
          
          # Load the JSON data
          with open(json_file_path, 'r') as json_file:
              json_data = json_file.read()
          
          # Add the custom fluid to CoolProp
          CP.add_fluids_as_JSON('HEOS', json_data)

However I get this error: 

Windows fatal exception: access violation

Main thread:
Current thread 0x00024668 (most recent call first):
  File "c:\users\kmirj\desktop\new refrigerant\overwrite.py", line 13 in <module>
  File "C:\Users\kmirj\anaconda3\Lib\site-packages\spyder_kernels\py3compat.py", line 356 in compat_exec
  File "C:\Users\kmirj\anaconda3\Lib\site-packages\spyder_kernels\customize\spydercustomize.py", line 473 in exec_code
  File "C:\Users\kmirj\anaconda3\Lib\site-packages\spyder_kernels\customize\spydercustomize.py", line 615 in _exec_file
  File "C:\Users\kmirj\anaconda3\Lib\site-packages\spyder_kernels\customize\spydercustomize.py", line 528 in runfile
  File "C:\Users\kmirj\AppData\Local\Temp\ipykernel_141672\3120520578.py", line 1 in <module>

Restarting kernel...
I am not sure what to do next also I am not a programmer so I would appreciate it if you guide me through in a step by step way !

**Comment by ibell** [comment]
What are the  contents of your R454B json file? You will almost certainly have to use REFPROP for mixtures. The routines in CoolProp are not that reliable, and many features are not present

**Reply by KoushaMirjavadi** [reply]
I tried to define it as a mixture of R32 and R1234yf 
My code is below: 

{
    "Name": "R454B",
    "CAS": "12345-67-8",
    "ASHRAE34": "A2L",
    "Tc": 351.25,
    "Pc": 5266900,
    "acentric": 0.296,
    "molemass": 62.61,
    "components": {
        "component1": {
            "name": "R1234yf",
            "mass_fraction": 0.311
        },
        "component2": {
            "name": "R32",
            "mass_fraction": 0.689
        }
    },
    "pseudo-pure": true
}

**Comment by ibell** [comment]
I am not sure where you got that JSON from, but that is not how mixtures work in CoolProp. In this case, you simply need to specify the components and their fractions when you instantiate the model. In the case of this mixture, you should check out the supporting information to this paper which contains a more accurate mixture model for 32/yf: https://pubs.aip.org/aip/jpr/article/52/1/013101/2881597 . There is an example file that shows how to call the model.

**Reply by KoushaMirjavadi** [reply]
Thank you so much for you help !
